Sources/StreamVideo/Utils/AudioRecorder/StreamAudioRecorder.swift

```diff
@@ -0,0 +1,180 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import AVFoundation
+import Combine
+import Foundation
+
+/// This class abstracts the usage of AVAudioRecorder, providing a convenient way to record and manage
+/// audio streams. It handles setting up the recording environment, starting and stopping recording, and
+/// publishing the average power of the audio signal. Additionally, it adjusts its behavior based on the
+/// presence of an active call, automatically stopping recording if needed.
+open class StreamAudioRecorder: @unchecked Sendable {
+
+    /// The builder used to create the AVAudioRecorder instance.
+    let audioRecorderBuilder: AVAudioRecorderBuilder
+
+    /// The audio session used for recording and playback.
+    let audioSession: AudioSessionProtocol
+
+    /// A private task responsible for setting up the recorder in the background.
+    private var setUpTask: Task<Void, Never>?
+
+    /// A cancellable used to schedule the update of audio meters.
+    private(set) var updateMetersTimerCancellable: AnyCancellable?
+
+    /// A PassthroughSubject that publishes the average power of the audio signal.
+    private var _metersPublisher: PassthroughSubject<Float, Never> = .init()
+
+    /// A public publisher that exposes the average power of the audio signal.
+    open private(set) lazy var metersPublisher: AnyPublisher<Float, Never> = _metersPublisher.eraseToAnyPublisher()
+
+    /// Indicates whether an active call is present, influencing recording behavior.
+    public var hasActiveCall: Bool = false {
+        didSet {
+            if !hasActiveCall {
+                Task {
+                    await stopRecording()
+                    try? audioSession.setActive(false, options: [])
```

Comment:
Should we log an error here?

Reply:
TBH it doesn't make much difference because either way our call has ended and this is only to mark the system's session (which any way will be deactivated after some time idle).

Reply:
In any way it won't hurt for sure :) Let me add it